### PR TITLE
VZ-5702: Update API, ingress, and istio pages with overrides documentation

### PR DESCRIPTION
**Pull request by mbenmclean**
This change updates VZ CRD API and the ingress and istio customize installation page to include information about install overrides.

### CODE REVIEW

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md [hunk 1]
```diff
@@ -75,6 +75,8 @@ spec:
 | --- | --- | --- | --- |
 | `enabled` | Boolean | If true, then AuthProxy will be installed. | No |
 | `kubernetes` | [AuthProxyKubernetes](#authproxy-kubernetes-configuration) | The Kubernetes resources than can be configured for AuthProxy. | No |
+| `monitorChanges` | Boolean | If false, Verrazzano updates will ignore any configuration changes to this component. Defaults to true. | No |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/Verrazzano/Verrazzano.md:78]: If false, Verrazzano -> If false, then Verrazzano (If... , then...); please fix globally.
Defaults to true -> Defaults to `true` (code font); code values should be in code font, please fix globally. (I made this change to the entire document in my last PR; I don't see that you pulled those changes into your PR).

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md [hunk 2]
```diff
@@ -75,6 +75,8 @@ spec:
...
+| `overrides` | [Override](#override) list | List of Overrides used to configure the component | No |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/Verrazzano/Verrazzano.md:79]: List of Overrides -> List of Overrides (overrides, not sure why you capitalized this term; tell me if there is a good reason for it.)
Please put a period at the end of each sentence; globally.

File: content/en/docs/setup/customizing/ingress.md [hunk 3]
```diff
@@ -6,10 +6,12 @@ Weight: 9
 draft: false
 ---
 
-Verrazzano uses NGINX for ingress to Verrazzano system components and Istio for application ingress.
-You can customize the NGINX and Istio ingress installation configurations using Helm overrides specified in the
-Verrazzano custom resource. For example, the following Verrazzano custom resource overrides the shape
-of an Oracle Cloud Infrastructure load balancer for both NGINX and Istio ingresses:
+Verrazzano uses NGINX for ingress to Verrazzano system components.
+You can customize the NGINX configuration using Overrides specified in the Verrazzano custom resource.
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/ingress.md:10]: Overrides  -> overrides
@mbenmclean This doc is called Customize Ingress, and has this description, "Customize Verrazzano NGINX and Istio ingress installation settings"
It seems to me that it describes only NGINX ingress and that there is a separate doc for Istio. Do you think that you should change the description to remove Istio from it and perhaps, even change the title to "Customize NGINX Ingress Controller"?

File: content/en/docs/setup/customizing/ingress.md [hunk 4]
```diff
@@ -6,10 +6,12 @@ Weight: 9
 draft: false
 ---
 
-Verrazzano uses NGINX for ingress to Verrazzano system components and Istio for application ingress.
-You can customize the NGINX and Istio ingress installation configurations using Helm overrides specified in the
-Verrazzano custom resource. For example, the following Verrazzano custom resource overrides the shape
-of an Oracle Cloud Infrastructure load balancer for both NGINX and Istio ingresses:
+Verrazzano uses NGINX for ingress to Verrazzano system components.
+You can customize the NGINX configuration using Overrides specified in the Verrazzano custom resource. 
+[Overrides](({{< relref "/docs/reference/API/Verrazzano/Verrazzano.md#override" >}})) can be specified as inline YAML
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/ingress.md:11]: This relref doesn't work; it has too many parentheses. It should look like this: 
({{< relref "/docs/reference/API/Verrazzano/Verrazzano.md#override" >}})

File: content/en/docs/setup/customizing/ingress.md [hunk 5]
```diff
@@ -6,10 +6,12 @@ Weight: 9
...
+or embedded in a ConfigMap or Secret.
+For example, the following Verrazzano custom resource overrides the autoscaling
+configuration for the NGINX ingress controller.
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/ingress.md:14]: NGINX ingress controller -> NGINX Ingress Controller (this is capitalized throughout the docs)

File: content/en/docs/setup/customizing/ingress.md [hunk 6]
```diff
@@ -21,28 +23,14 @@ spec:
...
-
-```
-controler:
-   service:
-     annotations:
-       service.beta.kubernetes.io/oci-load-balancer-shape: 10Mbps
-```
 For more information about setting component overrides, see [Customizing the Chart Before Installing](https://helm.sh/docs/intro/using_helm/#customizing-the-chart-before-installing).
+
+NGINX values available for customizing can be found in the `values.yaml` file in the [ingress-nginx Helm Chart](https://github.com/verrazzano/verrazzano/blob/master/platform-operator/thirdparty/charts/ingress-nginx/values.yaml)
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/ingress.md:36]: NGINX values available for customizing can be found at -> You can find NGINX values available for customizing at

File: content/en/docs/setup/customizing/istio.md [hunk 7]
```diff
@@ -6,6 +6,7 @@ weight: 6
 draft: false
 ---
 
+Verrazzano uses Istio to provide application ingress as well as to facilitate mTLS cluster communication.
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/istio.md:9]: mTLS -> mutual TLS authentication (mTLS)

File: content/en/docs/setup/customizing/istio.md [hunk 8]
```diff
@@ -6,6 +6,7 @@ weight: 6
...
 You can customize Verrazzano Istio component using settings in the Verrazzano custom resource.
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/istio.md:10]: customize Verrazzano Istio component -> customize the Verrazzano Istio component

File: content/en/docs/setup/customizing/istio.md [hunk 9]
```diff
@@ -58,3 +59,34 @@ The following example customizes a Verrazzano `prod` profile as follows:
                            - istio-egressgateway
                    topologyKey: kubernetes.io/hostname
  ```
+
+Istio can also be customized using [Overrides](({{< relref "/docs/reference/API/Verrazzano/Verrazzano.md#override" >}})).
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/istio.md:63]: Istio can also be customized using -> You can also customize Istio using
(Same problem with the relref that I mentioned previously, an extra set of parentheses.)

File: content/en/docs/setup/customizing/istio.md [hunk 10]
```diff
@@ -58,3 +59,34 @@ The following example customizes a Verrazzano `prod` profile as follows:
...
+          spec:
+            components:
+              ingressGateways:
+              - name: istio-ingressgateway
+                k8s:
+                  serviceAnnotations:
+                    service.beta.kubernetes.io/oci-load-balancer-shape: 10Mbps
+```
+
+When overriding values in the ingress gateway or the egress gateway it is necessary to specify the name,
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/istio.md:88]: or the egress gateway -> or the egress gateway, (comma)

File: content/en/docs/setup/customizing/istio.md [hunk 11]
```diff
@@ -58,3 +59,34 @@ The following example customizes a Verrazzano `prod` profile as follows:
...
+so either `istio-ingressgateway`, or `istio-egressgateway`.
+
+Verrazzano can accept any valid IstioOperator Custom Resource as an override. The IstioOperator API I can be reference
```
Comment (rosemarymarano) [inline content/en/docs/setup/customizing/istio.md:91]: The IstioOperator API I can be reference -> You can reference the IstioOperator API I

**Review by rosemarymarano — COMMENTED** [review]
Edit and comments.

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 1 above)
Comment (mbenmclean) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md:78]: fixed

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 2 above)
Comment (mbenmclean) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md:79]: I'm not sure if we want Overrides to be a proper noun

File: content/en/docs/setup/customizing/ingress.md (hunk 4 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/ingress.md:11]: My IDE seems to think it should have two parentheses. It rendered correctly too.

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 2 above)
Comment (rosemarymarano) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md:79]: Proper nouns are reserved for names of persons, places, and products. We do capitalize (though not consistently) Kubernetes objects and other objects that are capitalized in their own documentation. That said, it looked to me that overrides were inconsistently capitalized in your doc changes. If you want to capitalize it in reference to a component name (noun), then that's ok. Then, use lowercase overrides to refer to the verb.

File: content/en/docs/setup/customizing/ingress.md (hunk 4 above)
Comment (rosemarymarano) [reply content/en/docs/setup/customizing/ingress.md:11]: It didn't work in Hugo, which renders our docs. Try that and then please get back to me. Thank you.

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 2 above)
Comment (mbenmclean) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md:79]: Yeah I mean we do have this feature which I'm documenting called Overrides. I'm just not sure if we want to be referring to it in the docs as capital O Overrides, or just lowercase. Might be a @gilbode question.

File: content/en/docs/setup/customizing/ingress.md (hunk 5 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/ingress.md]: fixed

File: content/en/docs/setup/customizing/istio.md (hunk 7 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/istio.md]: fixed

File: content/en/docs/setup/customizing/istio.md (hunk 9 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/istio.md]: fixed

File: content/en/docs/setup/customizing/istio.md (hunk 11 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/istio.md]: fixed

File: content/en/docs/setup/customizing/ingress.md (hunk 6 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/ingress.md]: fixed

File: content/en/docs/setup/customizing/istio.md (hunk 10 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/istio.md]: fixed

File: content/en/docs/setup/customizing/istio.md (hunk 8 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/istio.md]: fixed

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 1 above)
Comment (rosemarymarano) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md]: @mbenmclean Not sure if you saw/fixed this part of my comment, "Defaults to true -> Defaults to `true` (code font); code values should be in code font, please fix globally. (I made this change to the entire document in my last PR; I don't see that you pulled those changes into your PR).

File: content/en/docs/setup/customizing/ingress.md (hunk 4 above)
Comment (mbenmclean) [reply content/en/docs/setup/customizing/ingress.md]: changed

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md [hunk 12]
```diff
@@ -75,6 +75,8 @@ spec:
 | --- | --- | --- | --- |
 | `enabled` | Boolean | If true, then AuthProxy will be installed. | No |
 | `kubernetes` | [AuthProxyKubernetes](#authproxy-kubernetes-configuration) | The Kubernetes resources than can be configured for AuthProxy. | No |
+| `monitorChanges` | Boolean | If false, then Verrazzano updates will ignore any configuration changes to this component. Defaults to true. | No |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/Verrazzano/Verrazzano.md:78]: Defaults to true -> Defaults to `true` (code font)

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md [hunk 13]
```diff
@@ -75,6 +75,8 @@ spec:
...
+| `overrides` | [Overrides](#overrides) list | List of Overrides for default `values.yaml` for component Helm Chart. If you pass in multiple Overrides that reference the same values, Verrazzano will use the final one. You can find all possible values [here]( {{< release_source_url path=platform-operator/helm_config/charts/verrazzano-authproxy/values.yaml >}} ) and invalid values will be ignored. | No |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/Verrazzano/Verrazzano.md:79]: `values.yaml` -> `values.yaml` file
Helm Chart -> Helm chart
If you pass in... (need If..., then)
Still not agreeing with capital Overrides. Was this specified by @gilbode ?
Please replicate my corrections globally.

**Review by rosemarymarano — COMMENTED** [review]
Still plenty to fix.

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 13 above)
Comment (rosemarymarano) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md]: OK.

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 12 above)
Comment (mbenmclean) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md]: fixed

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 13 above)
Comment (mbenmclean) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md]: I made the the other changes. Could you re-review and approve please if everything else looks good? Thank you.

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md [hunk 14]
```diff
@@ -75,8 +75,8 @@ spec:
...
-| `monitorChanges` | Boolean | If false, then Verrazzano updates will ignore any configuration changes to this component. Defaults to true. | No |
-| `overrides` | [Overrides](#overrides) list | List of Overrides for default `values.yaml` for component Helm Chart. If you pass in multiple Overrides that reference the same values, Verrazzano will use the final one. You can find all possible values [here]( {{< release_source_url path=platform-operator/helm_config/charts/verrazzano-authproxy/values.yaml >}} ) and invalid values will be ignored. | No |
+| `monitorChanges` | Boolean | If false, then Verrazzano updates will ignore any configuration changes to this component. Defaults to `true`. | No |
+| `overrides` | [Overrides](#overrides) list | List of Overrides for the default `values.yaml` file for component Helm Chart. If you pass in multiple Overrides that reference the same values, Verrazzano will use the final one. You can find all possible values [here]( {{< release_source_url path=platform-operator/helm_config/charts/verrazzano-authproxy/values.yaml >}} ) and invalid values will be ignored. | No |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/Verrazzano/Verrazzano.md:79]: for component Helm Chart -> for the component Helm chart (not sure why you are capitalizing chart)
If you pass in multiple..., Verrazzano -> If you pass in multiple..., then Verrazzano

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md [hunk 15]
```diff
@@ -89,8 +89,8 @@ spec:
...
 | `certificate` | [Certificate](#certificate) | The certificate configuration. | No |
-| `monitorChanges` | Boolean | If false, then Verrazzano updates will ignore any configuration changes to this component. Defaults to true. | No |
-| `overrides` | [Overrides](#overrides) list | List of Overrides for default `values.yaml` for component Helm Chart. If you pass in multiple Overrides that reference the same values, Verrazzano will use the final one. You can find all possible values [here]( {{< release_source_url path=platform-operator/thirdparty/charts/cert-manager/values.yaml  >}} ) and invalid values will be ignored. | No |
+| `monitorChanges` | Boolean | If false, then Verrazzano updates will ignore any configuration changes to this component. Defaults to `true`. | No |
+| `overrides` | [Overrides](#overrides) list | List of Overrides for the default `values.yaml` file file for component Helm Chart. If you pass in multiple Overrides that reference the same values, Verrazzano will use the final one. You can find all possible values [here]( {{< release_source_url path=platform-operator/thirdparty/charts/cert-manager/values.yaml  >}} ) and invalid values will be ignored. | No |
```
Comment (rosemarymarano) [inline content/en/docs/reference/API/Verrazzano/Verrazzano.md:93]: typo -> file file

**Review by rosemarymarano — COMMENTED** [review]
Not done yet. Please incorporate all the requested edits.

File: content/en/docs/reference/API/Verrazzano/Verrazzano.md (hunk 14 above)
Comment (mbenmclean) [reply content/en/docs/reference/API/Verrazzano/Verrazzano.md:79]: I was under the impression Chart was capitalized in the context of Helm Charts. Maybe not. From the helm documentation
https://helm.sh/docs/topics/charts/

**Review by rosemarymarano — APPROVED** [review]
YES!!!

**Pull request merged**